nuxeo-management-rest-api/src/main/java/org/nuxeo/rest/management/probes/ProbesObject.java

```diff
@@ -75,7 +82,11 @@ public ProbeInfo doGet(@PathParam("probeName") String probeName) {
     @Produces(MediaType.APPLICATION_JSON)
     @Path("{probeName}")
     public ProbeInfo launch(@PathParam("probeName") String probeName) {
-        return Framework.getService(ProbeManager.class).runProbe(probeName);
+        ProbeInfo probeInfo = Framework.getService(ProbeManager.class).runProbe(probeName);
+        if (probeInfo == null) {
+            throw new NuxeoException("No such probe: " + probeName, SC_NOT_FOUND);
+        }
+        return probeInfo;
```

Comment:
What do you think of factorizing these 2 methods?

Reply:
sure